src/audio/buffer.c

```diff
@@ -55,15 +56,19 @@ struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t align)
 		return NULL;
 	}
 
-	buffer_init(buffer, size, caps);
-	coherent_init_thread(buffer, c);
```

Comment:
@lgirdwood you mean those two `list_init()` lines below? I don't think you can do that. Those list headers hold buffers on component source and sink buffer lists respectively. You only add buffers to lists once when building pipelines and delete them when destroying pipelines. And you acquire and release buffers *many* times during operation between those two events. You cannot re-initialise those lists when acquiring buffers. Or did I misunderstand?

Reply:
I mean why is some buffer init done with buffer and some with buffer_c. It does not matter in this case as it was already mixed, but we should align it to use a single buffer ptr once the allocator is aligned with Zephyr.

Reply:
@lgirdwood Strictly speaking those list initialisations aren't needed at all - those aren't list heads but list items. But SOF does often initialise list items too to be able to use the `list_is_empty()` test on them to check if an item is on a list. Since cached alias pointers should only be used between acquire and release, we shouldn't keep them outside of those blocks. So, pointers to buffers should always be uncached. So if you want to use `list_init()` you have to do that on uncached. Or you'd have to open-code it as
```
/* Use a *cached* alias to assign an *uncached* pointer */
buffer_c->source_list.next = &buffer->source_list;
```
etc. Only a very small part of buffer initialisation happens via uncached access. That can be changed, but (1) this isn't critical - it's before the buffer is added to lists, so it doesn't have to be protected, and (2) since it's only initialisation, it isn't time-critical.

Also note, that buffer list headers are always accessed uncached because otherwise you wouldn't be able to use `list_for_item()` and other list helpers. You would have to open-code something like
```
item = head->list.next;
while (item != &head->list) {
	buf = container_of(item,...);
	buf_c = buffer_acquire(buf);
	/* do stuff */
	item = buf_c->source_list.next; /* or sink_list respectively */
	buffer_release(buf_c);
}
```
Whereas I do think that just using only those list headers uncached is safe, because they aren't changed at runtime. So even if cache gets written back to SRAM somewhere while a buffer is acquired, those headers will still have correct values.